### PR TITLE
feat(docs): Clarify gatsby-plugin-mdx requirements for gatsby-source-filesystem

### PR DESCRIPTION
**Pull request by yamalight**
## Description

Adds an example code and additional note explaining that mdx plugin needs to have filesystem pre-processing files.

## Related Issues

https://github.com/gatsbyjs/gatsby/issues/15594

### CODE REVIEW

**Comment by jonniebigodes** [comment]
@yamalight fine on my end. I'll leave to core team to review it and possibly merge it.

**Comment by marcysutton** [comment]
Ooh, great addition! The PR needs to be updated to use the current plugin version: `gatsby-plugin-mdx` and then it should be good to go.

Thanks for working on this!

**Comment by yamalight** [comment]
@marcysutton should be updated now I think 🤔

File: packages/gatsby-plugin-mdx/README.md [hunk 1]
```diff
@@ -55,14 +55,27 @@ After installing gatsby-plugin-mdx you can add it to your plugins list in your
...
+    `gatsby-plugin-mdx`,
+  ],
 }
 ```
 
 By default, this configuration will allow you to create pages
 with `.mdx` files in `src/pages` and will process any Gatsby nodes
 with Markdown media types into MDX content.
 
+Note that gatsby-mdx requires gatsby-source-filesystem to be present
```
Comment (johno) [inline packages/gatsby-plugin-mdx/README.md]:
```suggestion
Note that gatsby-plugin-mdx requires gatsby-source-filesystem to be present
```

File: packages/gatsby-plugin-mdx/README.md [hunk 2]
```diff
@@ -110,6 +123,20 @@ layout defined, even if it's imported manually using `import MDX from './thing.m
 // gatsby-config.js
 module.exports = {
   plugins: [
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        name: `pages`,
```
Comment (johno) [inline packages/gatsby-plugin-mdx/README.md]: Pages should be automatically sourced with MDX if there are any `.mdx` files. Were you not seeing this occur?

File: packages/gatsby-plugin-mdx/README.md [hunk 3]
```diff
@@ -55,14 +55,27 @@ After installing gatsby-plugin-mdx you can add it to your plugins list in your
...
+  ],
 }
 ```
 
 By default, this configuration will allow you to create pages
 with `.mdx` files in `src/pages` and will process any Gatsby nodes
 with Markdown media types into MDX content.
 
+Note that gatsby-mdx requires gatsby-source-filesystem to be present
+and configured to process your markdown files to correctly
```
Comment (johno) [inline packages/gatsby-plugin-mdx/README.md]:
```suggestion
and configured to process local markdown files in order to
```

File: packages/gatsby-plugin-mdx/README.md (hunk 2 above)
Comment (yamalight) [reply packages/gatsby-plugin-mdx/README.md]: Page were sourced, yes. But graphql node were not created for the pages, that's why I found it quite confusing, and that's why I think it's a good idea to include filesystem plugin in examples

**Comment by johno** [comment]
Thanks @yamalight!

**Pull request merged**